### PR TITLE
Use flags for create post

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
This PR is to clean up the diff on https://github.com/mattermost/mattermost/pull/28334

It uses flags for create post instead of arguments to allow a new flags to be added in the future without this much refactoring.

#### Ticket Link
NONE

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/channel.go [hunk 1]
```diff
@@ -2933,7 +2933,6 @@ func (a *App) SearchAllChannels(c request.CTX, term string, opts model.ChannelSe
 		PolicyID:                 opts.PolicyID,
 		IncludePolicyID:          opts.IncludePolicyID,
 		IncludeSearchById:        opts.IncludeSearchById,
-		ExcludeRemote:            opts.ExcludeRemote,
```
Comment (agnivade) [inline server/channels/app/channel.go:2936]: Is this related?

**Review by agnivade — APPROVED** [review]
Left a question

**Review by cpoile — APPROVED** [review]

File: server/channels/app/channel.go (hunk 1 above)
Comment (larkox) [reply server/channels/app/channel.go:2936]: Great catch! No, not related at all. May have been an error while merging.

**Comment by larkox** [comment]
@yasserfaraazkhan This should only need some smoke test around posting, that nothing seems broken. Probably just running the e2e tests is enough, but I leave that decision to you :smile:

**Review by yasserfaraazkhan — APPROVED** [review]
